input/kafkamdm/kafkamdm.go

```diff
@@ -93,263 +71,56 @@ func ConfigProcess(instance string) {
-		if err != nil {
-			log.Fatal(4, "kafkamdm: invalid offest format. %s", err)
-		}
-	}
 
-	offsetMgr, err = kafka.NewOffsetMgr(DataDir)
-	if err != nil {
-		log.Fatal(4, "kafka-mdm couldnt create offsetMgr. %s", err)
+	if consumerConf.MaxWaitMs == 0 {
+		log.Fatal(4, "kafkamdm: consumer-max-wait-time must be greater then 0")
 	}
-	brokers = strings.Split(brokerStr, ",")
-	topics = strings.Split(topicStr, ",")
-
-	config = sarama.NewConfig()
 
-	config.ClientID = instance + "-mdm"
-	config.ChannelBufferSize = channelBufferSize
-	config.Consumer.Fetch.Min = int32(consumerFetchMin)
-	config.Consumer.Fetch.Default = int32(consumerFetchDefault)
-	config.Consumer.MaxWaitTime = consumerMaxWaitTime
-	config.Consumer.MaxProcessingTime = consumerMaxProcessingTime
-	config.Net.MaxOpenRequests = netMaxOpenRequests
-	config.Version = sarama.V0_10_0_0
-	err = config.Validate()
-	if err != nil {
-		log.Fatal(2, "kafka-mdm invalid config: %s", err)
-	}
-	// validate our partitions
-	client, err := sarama.NewClient(brokers, config)
-	if err != nil {
-		log.Fatal(4, "kafka-mdm failed to create client. %s", err)
-	}
-	defer client.Close()
+	consumerConf.Topics = strings.Split(topicStr, ",")
+	consumerConf.ClientID = instance + "-mdm"
 
-	availParts, err := kafka.GetPartitions(client, topics)
-	if err != nil {
-		log.Fatal(4, "kafka-mdm: %s", err.Error())
-	}
-	log.Info("kafka-mdm: available partitions %v", availParts)
-	if partitionStr == "*" {
-		partitions = availParts
-	} else {
-		parts := strings.Split(partitionStr, ",")
-		for _, part := range parts {
-			i, err := strconv.Atoi(part)
-			if err != nil {
-				log.Fatal(4, "could not parse partition %q. partitions must be '*' or a comma separated list of id's", part)
-			}
-			partitions = append(partitions, int32(i))
-		}
-		missing := kafka.DiffPartitions(partitions, availParts)
-		if len(missing) > 0 {
-			log.Fatal(4, "kafka-mdm: configured partitions not in list of available partitions. missing %v", missing)
-		}
-	}
 	// record our partitions so others (MetricIdx) can use the partitioning information.
 	// but only if the manager has been created (e.g. in metrictank), not when this input plugin is used in other contexts
 	if cluster.Manager != nil {
-		cluster.Manager.SetPartitions(partitions)
-	}
-
-	// initialize our offset metrics
-	partitionOffset = make(map[int32]*stats.Gauge64)
-	partitionLogSize = make(map[int32]*stats.Gauge64)
-	partitionLag = make(map[int32]*stats.Gauge64)
-	for _, part := range partitions {
-		partitionOffset[part] = stats.NewGauge64(fmt.Sprintf("input.kafka-mdm.partition.%d.offset", part))
-		partitionLogSize[part] = stats.NewGauge64(fmt.Sprintf("input.kafka-mdm.partition.%d.log_size", part))
-		partitionLag[part] = stats.NewGauge64(fmt.Sprintf("input.kafka-mdm.partition.%d.lag", part))
+		consumer, err := kafka.NewConsumer(consumerConf)
+		if err != nil {
+			log.Fatal(2, "kafka-cluster failed to initialize consumer: %s", err)
+		}
+		if LogLevel < 2 {
```

Comment:
honestly i'm not sure, i've copied that from the many other locations in the code that do the same, i'm assuming it's for performance reasons to save a function call.
f.e.: https://github.com/grafana/metrictank/blob/master/api/dataprocessor.go?utf8=%E2%9C%93#L384

Reply:
yes, the function call is slow and allocates. IIRC because it uses interfaces.
wrapping the LogLevel optimizes it away, which is very useful in hot code paths (eg the ones that run for every message received), though this code doesn't look hot so it's not needed here.

logrus is much more efficient in non-printing calls, so when we switch over, we don't need loglevel checks in the callers anymore.

Reply:
to be more clear: please remove this. this code is not hot.

Reply:
removed